agrona/src/main/java/org/agrona/concurrent/status/AtomicCounter.java

```diff
@@ -211,10 +211,31 @@ public long increment()
 
     /**
      * Perform an atomic increment that is not safe across threads.
+     * <p>
+     * This method is identical to {@link #incrementRelease()} and that method should be used instead.
      *
      * @return the previous value of the counter
      */
     public long incrementOrdered()
+    {
+        return incrementRelease();
+    }
+
+    /**
+     * Perform a non-atomic increment.
+     * <p>
+     * It can result into lost updates due to race condition when called concurrently.
+     * <p>
+     * The load has plain memory semantics and the store has release memory semantics.
+     * <p>
+     * The typical use-case is when there is a single writer thread and one or more reader threads.
+     * <p>
+     * This method will outperform the {@link #increment()}. So if there is just a single mutator thread, and
+     * one or more reader threads, then it is likely you will prefer this method.
+     *
+     * @return the previous value of the counter
+     */
+    public long incrementRelease()
```

Comment:
Use `org.agrona.UnsafeApi#getAndAddLongRelease` to ensure that increment is atomic.

Reply:
The documentation is clear that the increment isn't atomic. Making it atomic is going to lead to a performance degradation. On the X86, it will result into a lock prefixed instruction which is significantly slower using an x86 plain load/store.

Reply:
All good then.